plugin-flex-ts-template-v2/src/feature-library/teams-view-enhancements/flex-hooks/components/AgentActivityIcon/AgentActivityIcon.tsx

```diff
@@ -0,0 +1,41 @@
+import { Icon } from '@twilio/flex-ui';
+import * as React from 'react';
+import { Tooltip } from '@twilio-paste/core/tooltip';
+
+import { AgentActivity, Heading } from './AgentActivityIcon.Components';
+
+interface ActivityConfig {
+  color: string;
+  icon: string;
+}
+
+interface ComponentProps {
+  activityName: string;
+  activityConfig: {
+    activities: {
+      [key: string]: ActivityConfig;
+    };
+    other: ActivityConfig;
+  };
+}
+
+const AgentActivityIcon = (props: ComponentProps) => {
+  const { activityName, activityConfig } = props;
+  let bgColor = activityConfig.other.color;
+  let icon = activityConfig.other.icon;
```

Comment:
I think we need to do some checks here if activityConfig is incomplete or missing. If someone removed the data tiles feature, it would be missing, so some sort of fallback should be in place.

Reply:
Another option rather than doing it here, would be to do the null checks within config.ts.

Reply:
so in case "metrics_data_tiles" can not be found in the ui_attributes, it will use that configuration